pkg/controller/controller_test.go

```diff
@@ -63,83 +59,139 @@ var _ = Describe("Controller", func() {
+		podSource = k8stesting.NewFakeControllerSource()
+	})
+
 	AfterEach(func() {
 		close(stop)
 	})
 
-	tests := []testT{
-		{
-			descr:       "pvc, endpoint, blank ns: controller creates importer pod",
-			ns:          "", // blank used originally in these unit tests
-			name:        "test-pvc",
-			qops:        opAdd,
-			annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
-			expectError: false,
-		},
-		{
-			descr:       "pvc, endpoint, non-blank ns: controller creates importer pod",
-			ns:          "ns-a",
-			name:        "test-pvc",
-			qops:        opAdd,
-			annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
-			expectError: false,
-		},
-		{
-			descr:       "pvc, blank endpoint: controller does not create importer pod",
-			ns:          "",
-			name:        "test-pvc",
-			qops:        opAdd,
-			annotations: map[string]string{},
-			expectError: true,
-		},
-		{
-			descr:       "updated pvc should process",
-			ns:          "ns-a",
-			name:        "test-pvc",
-			qops:        opUpdate,
-			annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
-			expectError: false,
-		},
-		{
-			descr:       "updated pvc should not process based on annotation AnnImportPod indicating already been processed",
-			ns:          "ns-a",
-			name:        "test-pvc",
-			qops:        opUpdate,
-			annotations: map[string]string{AnnEndpoint: "http://www.google.com", AnnImportPod: "importer-test-pvc"},
-			expectError: true,
-		},
-	}
+	Context("when parsing pvc annotations", func() {
 
-	for _, test := range tests {
-		ns := test.ns
-		pvcName := test.name
-		ops := test.qops
-		annotations := test.annotations
-		fullname := pvcName
-		if len(ns) > 0 {
-			fullname = fmt.Sprintf("%s/%s", ns, pvcName)
+		type testT struct {
+			descr       string
+			ns          string
+			name        string // name of test pvc
+			qops        operation
+			annotations map[string]string
+			expectError bool
 		}
-		exptPod := fmt.Sprintf("importer-%s-", pvcName)
-		exptErr := test.expectError
-
-		It(test.descr, func() {
-			By(fmt.Sprintf("creating in-mem pvc %q with endpt anno=%q", fullname, annotations))
-			pvc := createInMemPVC(ns, pvcName, annotations)
-			By("Creating the controller")
-			setUpInformer(pvc, ops)
-			controller.ProcessNextPvcItem()
-			//controller.ProcessNextPodItem()
-			By("checking if importer pod is present")
-			pod, err := getImporterPod(fakeClient, ns, exptPod)
-			if exptErr {
-				Expect(err).ToNot(BeNil(), fmt.Sprintf("importer pod %s... should not exist\n", exptPod))
-			} else {
-				Expect(err).To(BeNil(), fmt.Sprintf("importer pod: %v\n", err))
-				Expect(pod).ToNot(BeNil(), fmt.Sprintf("importer pod %q missing", exptPod))
-				Expect(pod.GenerateName).To(HavePrefix(exptPod))
+
+		tests := []testT{
+			{
+				descr:       "pvc, endpoint, blank ns: controller creates importer pod",
+				ns:          "", // blank used originally in these unit tests
```

Comment:
I _think_ @screeley44  and I agreed we don't need or want to test blank namespaces, so these cases can be deleted (and any special case code to construct the key).

Reply:
that's fine, but i'd prefer to do that in a separate PR